src/wiki/plugins/editsection/markdown_extensions.py

```diff
@@ -0,0 +1,117 @@
+            'headers': None,     # List of FindHeader, all headers with there positions
+            'location': None,    # To be extracted header
+            'header_id': None    # Header text ID of the to be extracted header
+        }
+        super(EditSectionExtension, self).__init__(**kwargs)
+
+    def extendMarkdown(self, md, md_globals):
+        ext = EditSectionProcessor(md)
+        ext.config = self.config
+        md.treeprocessors.add('editsection', ext, '_end')
+
+
+def get_header_id(header):
+    header_id = ''.join(w[0] for w in re.findall("\w+", header))
+    if not len(header_id):
+        return '_'
+    return header_id
+
+
+class EditSectionProcessor(Treeprocessor):
+    def locate_section(self, node):
+        cur_pos = [0] * self.level
+        last_level = 0
+        cur_header = -1
+        sec_level = -1
+        sec_start = -1
+
+        for child in node.getchildren():
+            match = self.HEADER_RE.match(child.tag.lower())
+            if not match:
+                continue
+
+            level = int(match.group(1))
+
+            # Find current position in headers
+            cur_header += 1
+            while (cur_header < len(self.headers) and
+                   not self.headers[cur_header].sure_header and child.text != self.headers[cur_header].header):
+                cur_header += 1
+            if cur_header >= len(self.headers):
+                return None
+
+            # End of the searched section found?
+            if level <= sec_level:
+                return sec_start, self.headers[cur_header].start
+
+            for l in range(level, last_level):
+                cur_pos[l] = 0
+            cur_pos[level - 1] += 1
+            last_level = level
+
+            location = '-'.join(map(str, cur_pos))
+            if location != self.location:
+                continue
+
+            # Found section start. Check if the header id text is still correct.
+            if get_header_id(child.text) != self.header_id:
+                return None
+
+            # Correct section start found. Search now for the section end.
+            sec_level = level
+            sec_start = self.headers[cur_header].start
+
+        if sec_start >= 0:
+            return sec_start, 9999999
+        return None
+
+    def add_links(self, node):
+        cur_pos = [0] * self.level
+        last_level = 0
+
+        for child in node.getchildren():
+            match = self.HEADER_RE.match(child.tag.lower())
+            if not match:
+                continue
+
+            level = int(match.group(1))
+            for l in range(level, last_level):
+                cur_pos[l] = 0
+            cur_pos[level - 1] += 1
+            last_level = level
+            location = '-'.join(map(str, cur_pos))
+            header_id = get_header_id(child.text)
+
+            # Insert link to allow editing this section
+            link = etree.SubElement(child, 'a')
+            link.text = "[edit]"
```

Comment:
This should be marked for translation!

Reply:
And I think perhaps even theme-able. I was thinking that an obvious alternative would be to change it into a pencil-icon. But I guess it might be theme-able with the parent `article-edit` class.

Reply:
Oops, fixed.

I added now a new setting, which defaults to a translated "[edit]".